src/test/scheduler/scheduler_test.cpp

```diff
@@ -140,7 +142,8 @@ TEST_F(SchedulerTest, LinearDependenciesWithScheduler) {
 }
 
 TEST_F(SchedulerTest, MultipleDependenciesWithScheduler) {
-  CurrentScheduler::set(std::make_shared<NodeQueueScheduler>(Topology::create_fake_numa_topology(8, 4)));
+  Topology::current().create_fake_numa_topology(8, 4);
```

Comment:
I find that a bit unintuitive to read, I would expect a `create_*` to return the new topology, not to reset the current topology to it. `Topology::current().init_fake_numa_topology(8, 4);`, maybe?

Reply:
I like `init_*` :+1: